### PR TITLE
net-mail/notmuch: Bump to version 0.30

### PR DESCRIPTION
**Pull request by rseichter**
Feature release 0.30, see [upstream news](https://notmuchmail.org/news/). This ebuild adds support for Python 3.8 and for the new Python CFFI bindings (package name "notmuch2").

### CODE REVIEW

File: net-mail/notmuch/notmuch-0.30.ebuild [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+DEPEND="${CDEPEND}
+	test? (
+		>=app-editors/emacs-${NEED_EMACS}:*[libxml2]
+		app-misc/dtach
+		sys-devel/gdb
+		crypt? ( app-crypt/gnupg dev-libs/openssl )
+	)
+	valgrind? ( dev-util/valgrind )
+"
+RDEPEND="${CDEPEND}
```
Comment (juippis) [inline net-mail/notmuch/notmuch-0.30.ebuild:58]: Here too.

File: net-mail/notmuch/notmuch-0.30.ebuild [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+		virtual/perl-Pod-Parser
+	)
+	nmbug? ( dev-vcs/git )
+"
+
+DOCS=( AUTHORS NEWS README )
+DOC_CONTENTS="There are a few backward-incompatible changes between
+notmuch version 0.18 < x <= 0.18. Please consult the NEWS file (0.18
+section) before first use."
+MY_LD_LIBRARY_PATH="${WORKDIR}/${P}/lib"
```
Comment (juippis) [inline net-mail/notmuch/notmuch-0.30.ebuild:78]: This is used only once, I think you're better off defining the path in `src_test`.

File: net-mail/notmuch/notmuch-0.30.ebuild [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+SITEFILE="50${PN}-gentoo.el"
+
+bindings() {
+	local rc=0
+	if use python; then
+		pushd bindings/python-cffi || die
+		"$@"
```
Comment (juippis) [inline net-mail/notmuch/notmuch-0.30.ebuild:85]: `${@}`

File: net-mail/notmuch/notmuch-0.30.ebuild [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+		rc=$?
```
Comment (juippis) [inline net-mail/notmuch/notmuch-0.30.ebuild:86]: `${?}`

File: net-mail/notmuch/notmuch-0.30.ebuild [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+		popd || die
+	fi
+	return $rc
```
Comment (juippis) [inline net-mail/notmuch/notmuch-0.30.ebuild:89]: `${rc}`
https://projects.gentoo.org/qa/policy-guide/ebuild-format.html#pg0101

File: net-mail/notmuch/notmuch-0.30.ebuild [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+src_prepare() {
+	local _deps=""
+	default
+
+	# Python bindings
+	bindings distutils-r1_src_prepare
+	mv contrib/notmuch-mutt/README contrib/notmuch-mutt/README-mutt || die
+
+	# Ensure that a new Makefile.config will be generated
+	rm -f Makefile.config || die
```
Comment (juippis) [inline net-mail/notmuch/notmuch-0.30.ebuild:113]: What's the point of `-f` here, is this generated randomly? FYI I took the `-f` out and this file didn't exist. Is this line just a remnant?

File: net-mail/notmuch/notmuch-0.30.ebuild [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+		elog ""
+		elog "  source ${_rcfile}"
+	fi
+}
+
+pkg_postinst() {
+	use emacs && elisp-site-regen
+	if has_version '<net-mail/notmuch-0.18'; then
+		FORCE_PRINT_ELOG=1 readme.gentoo_print_elog
+	fi
```
Comment (juippis) [inline net-mail/notmuch/notmuch-0.30.ebuild:207]: This looks redundant honestly, allows you to remove the eclass inherit and `DOC_CONTENT` variable here.

**Comment by juippis** [comment]
Oh and test phase passed... but failed on me. Does it work for you? I wouldn't consider that critical though.

**Comment by rseichter** [comment]
Tests are looking good for me:
```
Notmuch test suite complete.
All 1150 tests behaved as expected (4 expected failures).
All tests in 2 files skipped.
>>> Completed testing net-mail/notmuch-0.30
```

File: net-mail/notmuch/notmuch-0.30.ebuild [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+	python? ( ${PYTHON_DEPS} dev-python/cffi )
+"
+DEPEND="${CDEPEND}
+	test? (
+		>=app-editors/emacs-${NEED_EMACS}:*[libxml2]
+		app-misc/dtach
+		sys-devel/gdb
+		crypt? ( app-crypt/gnupg dev-libs/openssl )
+	)
+	valgrind? ( dev-util/valgrind )
```
[the remark this answers is not in the document]
Comment (rseichter) [reply net-mail/notmuch/notmuch-0.30.ebuild:62]: It is only used during testing.

File: net-mail/notmuch/notmuch-0.30.ebuild (hunk 3 above)
Comment (rseichter) [reply net-mail/notmuch/notmuch-0.30.ebuild]: One of many vestiges of the previous author's ebuild versions. 😉

File: net-mail/notmuch/notmuch-0.30.ebuild (hunk 6 above)
Comment (rseichter) [reply net-mail/notmuch/notmuch-0.30.ebuild]: I removed the statement. The current upstream distribution archive indeed does not contain a `Makefile.config`. Another bit of vestigal code.

**Pull request closed without merging**